domain/port/state/state.go

```diff
@@ -243,6 +244,42 @@ AND unit_endpoint.endpoint = $endpointName.endpoint
 	return decodedPortRanges, nil
 }
 
+// SetUnitPorts sets open ports for the endpoints of a given unit.
+func (st *State) SetUnitPorts(
+	ctx context.Context, unitName string, openPorts network.GroupedPortRanges,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Errorf("getting DB: %w", err)
+	}
+
+	endpoints := make([]string, len(openPorts))
+	i := 0
+	for endpoint := range openPorts {
+		endpoints[i] = endpoint
+		i++
+	}
+
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		unitUUID, err := st.getUnitUUID(ctx, tx, unitName)
+		if err != nil {
+			return errors.Errorf("getting uuid of unit %q: %w", unitName, err)
+		}
+
+		endpoints, err := st.ensureEndpoints(ctx, tx, unitUUID, endpoints)
+		if err != nil {
+			return errors.Errorf("ensuring endpoints exist for unit %q: %w", unitName, err)
+		}
+
+		err = st.openPorts(ctx, tx, openPorts, nil, endpoints)
```

Comment:
This is a model-migration-only method right? Setting `currentOpenedPorts` to nil here could lead to weirdness if it is already open.

It would be nice if this were clearer. I see in the `secrets` domain, for instance, the service has a `ImportSecrets` method, in it's own `exportimport.go` file

Reply:
Also, does this function meaningfully differ from `UpdateUnitPorts` with empty `closePorts`?

A different method on the service makes sense, but could we re-use code on the state side?

Reply:
It does in that it uses the unit name rather than the unit uuid. Though tbh thats the only real reason its a different method.

Reply:
In that case I would rather use the application service to get the uuid, and use `UpdateUnitPorts`